src/main/scala/jigg/util/JSONUtil.scala

```diff
@@ -116,6 +120,64 @@ object JSONUtil {
   }
 
   def writeToJSON[T <: Node](x: T, os: BufferedWriter): Unit = {
-    os.write(toJSON(x).toString)
+    os.write(toJSON(x))
+  }
+
+  def parseJSON(json: String): JValue = parse(json)
+
+  def loadJSONFile(name: String): JValue = {
+    val sb = new StringBuilder
+    for (line <- Source.fromFile(name).getLines){
+      sb.append(line)
+    }
+    parse(sb.toString)
+  }
+
+  def toXML(json: JValue): Node = {
+    implicit val formats = DefaultFormats
+    val jsonList = json.extract[Map[String, Any]].toList
+    generateXML(jsonList)
+  }
+
+  private def generateXML(x:List[(String, Any)]): Node = {
+    var tagString = ""
+    var textString = ""
+    val attributeList = new ArrayBuffer[(String, String)]
+    val nodeSeq = new ArrayBuffer[Node]
+    var hasChildFlag = false
+    var hasTextFlag = false
+
+    for (i <- x){
+      if (i._1 == ".tag")
+        tagString = i._2.toString
+      else if (i._1 == "text"){
+        hasTextFlag = true
+        textString = i._2.toString
+      }
+      else if (i._1 == ".child"){
+        hasChildFlag = true
+        for (j <- i._2.asInstanceOf[List[Map[String, Any]]])
+          nodeSeq += generateXML(j.toList)
+      }
+      else
+        attributeList += (i._1.toString -> i._2.toString)
+    }
+
+    val node = hasChildFlag match {
+      case true => {
+        hasTextFlag match {
+          case true => <xml>{textString}</xml>.copy(label = tagString)
+          case false => <xml></xml>.copy(label = tagString)
+        }
+      }
+      case false => <xml/>.copy(label = tagString)
```

Comment:
Unnecessary to catch cases where `hasTextFlag = true` but `hasChildFlag = false`? (seems theoretically possible)